bessctl/conf/testing/run_module_tests.py

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env python
+
+# Copyright (c) 2017, UC Berkeley
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+# list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+# this list of conditions and the following disclaimer in the documentation
+# and/or other materials provided with the distribution.
+#
+# * Neither the names of the copyright holders nor the names of their
+# contributors may be used to endorse or promote products derived from this
+# software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+from __future__ import print_function
+
+import fnmatch
+import glob
+import os
+import shlex
+import subprocess
+import sys
+import unittest
+
+this_dir = os.path.dirname(os.path.realpath(__file__))
+bessctl = os.path.join(this_dir, '../../bessctl')
+test_dir = os.path.join(this_dir, 'module_tests')
+
+
+class CommandError(subprocess.CalledProcessError):
+
+    '''Identical to CalledProcessError, except it also shows the output'''
+
+    def __str__(self):
+        return '%s\n%s' % (super(CommandError, self).__str__(), self.output)
+
+
+def run_cmd(cmd):
+    args = shlex.split(cmd)
+    try:
+        ret = subprocess.check_call(args, stderr=subprocess.STDOUT)
+    except subprocess.CalledProcessError as e:
+        raise CommandError(e.returncode, e.cmd, e.output)
+
+
+def main():
+    any_failure = 0
+
+    try:
+        run_cmd('%s daemon start' % bessctl)
+    except CommandError:
+        raise Exception('bess daemon could not start')
+
+    for file_name in glob.glob(os.path.join(test_dir, "*.py")):
```

Comment:
Wouldn't this mean that the feature of testing just a single module disappears?  I've found that very useful.

Reply:
I propose a new one! Please check a `assertBessAlive()` in `test_util.py` and see sample code in module test code.

Reply:
Maybe I misunderstand, but I guess he meant the configurable `TEST_PATH` and `TEST_NAME` environment variable  here, not about testing "a single module" vs "a chain of modules"

Reply:
Oh, sorry. it was 'my' misunderstand. @nemethf, you can simply run a single test within `bessctl`. For example, simply do `run testing/xxx.py`, then the test results will show up on the shell.

Reply:
Oh, that's great!  (And sorry for the noise.)